pkg/providers/docker/validator.go

```diff
@@ -0,0 +1,14 @@
+package docker
+
+import (
+	"fmt"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+)
+
+func validateControlPlaneEndpoint(clusterSpec *cluster.Spec) error {
```

Comment:
For other reviewers: the idea is to call this from the controller when validating the docker clusters during create